moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/XMLBindingContextFactory.java

```diff
@@ -30,12 +32,48 @@ public XMLBindingContextFactory() {
 
     @Override
     public JAXBContext createContext(Class<?>[] types, Map<String, ?> map) throws JAXBException {
-        return JAXBContextFactory.createContext(types, map);
+        Object value = map != null ? map.get(JAXBContextProperties.MOXY_FACTORY) : null;
+        // Property vas not set, use default factory
+        if (value == null) {
+            return JAXBContextFactory.createContext(types, map);
+        }
+        // Handle valid String properties
+        if (String.class.isInstance(value)) {
+            switch ((String) value) {
+                case JAXBContextProperties.Factory.DEFAULT:
+                    return JAXBContextFactory.createContext(types, map);
+                case JAXBContextProperties.Factory.DYNAMIC:
+                    return DynamicJAXBContextFactory.createContext(types, (Map<String, Object>) map);
+                default:
+                    throw new JAXBException(String.format("Property eclipselink.moxy.factory value \"%s\" is invalid", value));
+            }
+        // Non String values are invalid
+        } else {
+            throw new JAXBException(String.format("Property eclipselink.moxy.factory value \"%s\" is invalid", value.toString()));
+        }
```

Comment:
what about sth like:
```
    @Override
    public JAXBContext createContext(Class<?>[] types, Map<String, ?> map) throws JAXBException {
        @SuppressWarnings({"unchecked"})
        Map<String, Object> opts = map != null ? (Map<String, Object>) map : Collections.emptyMap();
        String factory = (String) opts.getOrDefault(JAXBContextProperties.MOXY_FACTORY, JAXBContextProperties.Factory.DEFAULT);
        switch (factory) {
            case JAXBContextProperties.Factory.DEFAULT:
                return JAXBContextFactory.createContext(types, opts);
            case JAXBContextProperties.Factory.DYNAMIC:
                return DynamicJAXBContextFactory.createContext(types, opts);
            default:
                throw new JAXBException(String.format("Property eclipselink.moxy.factory value \"%s\" is invalid", factory));
        }
    }
```

Reply:
note that the message should come from resource bundle, so sth like:
```
throw new JAXBException(ExceptionLocalization.buildMessage( somekey, new Object {factory}));
```

Reply:
Current version is 100% foolproof and won't throw classcast exception when property value is not String. But it's up to you. If you want to simplify it a bit for such a price, I can do that.